src/AspNetCore/GraphQLHttpMiddleware.cs

```diff
@@ -64,6 +69,9 @@ private bool IsGraphQlRequest(HttpContext context)
                 _.UserContext = _options.BuildUserContext?.Invoke(context);
                 _.ExposeExceptions = _options.ExposeExceptions;
                 _.ValidationRules = _options.ValidationRules.Concat(DocumentValidator.CoreRules()).ToList();
+                _documentListeners
+                    .ToList()
```

Comment:
This will allocate a new list. Less allocations is better.

Reply:
okey! I fix!